fastestimator/dataset/interleave_dataset.py

```diff
@@ -0,0 +1,271 @@
+# Copyright 2023 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import random
+from typing import Any, Dict, Iterable, List, Mapping, Optional, Union
+
+from torch.utils.data import Dataset
+
+from fastestimator.dataset.dataset import DatasetSummary, FEDataset
+from fastestimator.util.base_util import to_list, warn
+from fastestimator.util.traceability_util import traceable
+
+
+@traceable()
+class InterleaveDataset(FEDataset):
+    """A Dataset class that can allow for a step-wise interleaving of multiple datasets. This will be useful for
+    training multi-task models when we vary dataset used on a per-step basis.
+
```

Comment:
```suggestion
    """A Dataset class that can allow for a step-wise interleaving of multiple datasets.
    
    This will be useful for training multi-task models when we vary dataset used on a per-step basis.

```